editor/inserter/menu.js

```diff
@@ -58,37 +56,66 @@ class InserterMenu extends Component {
 		} );
 	}
 
-	selectBlock( name ) {
+	selectBlock( blockType ) {
 		return () => {
-			this.props.onSelect( name );
+			this.props.onSelect( blockType.name );
+			recordUsage( blockType );
```

Comment:
Sounds sensible, I have a branch started locally to explore storage options, so I'll carry on work here and make sure it's easy to switch over to use API calls.